Bug introduced by #4709

When testing workers-rs against main I was getting this build error:

```
error[E0596]: cannot borrow data in dereference of `Pin<&mut TimeoutFuture>` as mutable
   --> /Users/gbedford/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gloo-timers-0.3.0/src/future.rs:104:31
    |
104 |         Future::poll(Pin::new(&mut self.rx), cx).map(|t| t.unwrap_throw())
    |                               ^^^^^^^^^^^^ cannot borrow as mutable
    |
    = help: trait `DerefMut` is required to modify through a dereference, but it is not implemented for `Pin<&mut TimeoutFuture>`

error[E0596]: cannot borrow data in dereference of `Pin<&mut IntervalStream>` as mutable
   --> /Users/gbedford/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gloo-timers-0.3.0/src/future.rs:160:36
    |
160 |         Stream::poll_next(Pin::new(&mut self.receiver), cx)
    |                                    ^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
    |
    = help: trait `DerefMut` is required to modify through a dereference, but it is not implemented for `Pin<&mut IntervalStream>`

```

bisecting the build, shows this is caused by #4709.

We need to make sure to fix or revert before the next release.

--- a/src/closure.rs
+++ b/src/closure.rs
@@ -248,9 +248,13 @@ extern "C" {
 /// ```
 pub struct Closure<T: ?Sized> {
     js: JsClosure,
-     // careful: must be Box<T> not just T because unsized PhantomData
-     // seems to have weird interaction with Pin<>
+    // careful: must be Box<T> not just T because unsized PhantomData
+    // seems to have weird interaction with Pin<>
     _marker: PhantomData<Box<T>>,
+}
+
+fn _assert_compiles<T>(mut pin: core::pin::Pin<&mut Closure<T>>) {
+    let _ = &mut *pin;
 }
 
 impl<T> Closure<T>

